packages/react-input/src/components/Input/useInput.ts

```diff
@@ -29,12 +30,26 @@ export const useInput = (props: InputProps, ref: React.Ref<HTMLElement>): InputS
       contentBefore: 'span',
       contentAfter: 'span',
     },
-    input: resolveShorthand(input, { required: true }),
-    contentAfter: resolveShorthand(contentAfter),
-    contentBefore: resolveShorthand(contentBefore),
-    root: getNativeElementProps('span', {
-      ref,
-      ...props,
+    input: resolveShorthand(props.input, {
+      required: true,
+      defaultProps: {
+        type: 'text',
+        ref,
+        ...nativeProps.primary,
+      },
+    }),
+    contentAfter: resolveShorthand(props.contentAfter),
+    contentBefore: resolveShorthand(props.contentBefore),
+    root: resolveShorthand(props.root, {
+      required: true,
+      defaultProps: nativeProps.root,
     }),
   };
+
+  state.input.onChange = useEventCallback(ev => {
+    const newValue = ev.target.value;
+    onChange?.(ev, { value: newValue });
+  });
```

Comment:
Out of curiosity, what's the value in extending `onChange` here? This isn't incorporating any internal state or anything that the user couldn't get with a native `onChange` handler.

Reply:
There's an RFC establishing that this should be the standard pattern for onChange handlers. https://github.com/microsoft/fluentui/blob/master/rfcs/convergence/event-handlers-arguments.md